activestorage/lib/active_storage/attached.rb

```diff
@@ -8,8 +8,12 @@ module ActiveStorage
   class Attached
     attr_reader :name, :record
 
-    def initialize(name, record)
-      @name, @record = name, record
+    def initialize(name, record, configuration)
+      @name, @record, @configuration = name, record, configuration
+    end
+
+    def find_variant_by_name(variant_name)
+      @configuration[:defined_variants][variant_name.to_sym]
     end
```

Comment:
Convenience/Readability method to find a defined variant. Emphasizes consistently storing and looking up names as `Symobl`s